torch/csrc/jit/ir/ir.cpp

```diff
@@ -870,6 +870,17 @@ void Value::replaceAllUsesAfterNodeWith(const Node* node, Value* newValue) {
       uses_.end());
 }
 
+Value* Value::findFirstUseInBlock() {
+  const Block* this_block = this->node()->owningBlock();
+  for (const Use& u : uses()) {
+    Value* comparison = u.user->inputs_[u.offset];
+    if (comparison->node()->owningBlock() == this_block) {
+      return comparison;
+    }
+  }
+  return this;
+}
+
```

Comment:
This part of logic seems to rely on `uses()` to return all uses in same order as user nodes appear in a block. Based on my rough reading of implementation, this is not guaranteed. Maybe I am missing something.  Could you clarify?

Reply:
Oops...I assumed that `uses()` would indeed return all uses in the same order as the nodes appear in the block. I'm not familiar with this API myself (I'm using it for the first time here), and it looked to me like `Value::replaceFirstUseWith` ([here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/ir/ir.cpp#L845)) also relies on that being true. Should I also fix `replaceFirstUseWith`? Or am I somehow misreading it? Thanks!